src/server/api/schema.js

```diff
@@ -1197,6 +1199,43 @@ const rootMutations = {
       }
 
       return jobId;
+    },
+    createTag: async (_, { organizationId, tagData }, { user }) => {
+      await accessRequired(user, organizationId, "SUPERVOLUNTEER");
+      const tagInstance = new Tag({
+        organization_id: organizationId,
+        name: tagData.name,
+        group: tagData.group,
+        description: tagData.description,
+        is_deleted: false
+      });
+      const newTag = await tagInstance.save();
+      return newTag;
+    },
+    editTag: async (_, { organizationId, tagData, id }, { user }) => {
+      await accessRequired(user, organizationId, "SUPERVOLUNTEER");
+      const tagUpdates = {
+        name: tagData.name,
+        group: tagData.group,
+        description: tagData.description,
+        is_deleted: tagData.isDeleted,
+        organization_id: organizationId
+      };
+
+      await r
+        .knex("tag")
+        .where("id", id)
+        .update(tagUpdates);
+
+      return { id, ...tagUpdates };
+    },
+    deleteTag: async (_, { organizationId, id }, { user }) => {
+      await accessRequired(user, organizationId, "SUPERVOLUNTEER");
+      await r
+        .knex("tag")
+        .where("id", id)
+        .update({ is_deleted: true });
```

Comment:
Maybe I missed something in the issue conversation, but why don't we actually delete the tags? Is it to avoid issues with cascading deletes in the `tag_campaign_contact` table?

Reply:
@ibrand  gave me part of the explanation for this, but I wasn't totally clear on it and there may be more that I'm not aware of. Ilona?

Reply:
Hi @aschneit and @azuzunaga thanks for the ping :)

I encouraged a soft delete because if tags are soft deleted but still exist attached to other things (like canned responses) then no other data will be messed up. Secondarily, I figure an admin could accidentally delete one and want it back.

I think that first situation is a lot more important because it could lead to some bad data situations for a tag to no longer exist but a lot of the data export has tags already in it.

Reply:
Thanks for the explanation @ibrand! Makes sense to me.